### PR TITLE
Stamina indicator HUD for humans

### PR DESCRIPTION
**Pull request by jacklecroy**
## About The Pull Request

Adds a stamina indicator HUD for human mobs. 

## Why It's Good For The Game

It is confusing and unintuitive, especially to newer players, to combine stamina and health into the same HUD element. It can be hard to distinguish between toxin damage and stamina damage. This change still keeps some uncertainty. The stamina and health indicators are both only accurate to the nearest 20.

## Changelog

:cl: That REALLY Good Soda Flavor
imageadd: Added a stamina indicator to the human mob's HUD.
/:cl:

Sprites by @Kierany9 in 2016.

### CODE REVIEW

**Comment by theta006** [comment]
this is good lets just pray no one makes running take stamina

**Comment by DrDiasyl** [comment]
I don't really that it's named stamina damage tbh. Maybe we should rename it into pain? I actually want to do that and make a fluff system where when you're stamina damaged enough, your screen starts to get red and your character can moan, scream or cry.

**Comment by Mothblocks** [comment]
i don't know why, but i feel off about this

**Comment by DrDiasyl** [comment]
> i don't know why, but i feel off about this

Probably because it reminds you of something.. same.. I think I seen this lighting symbol somewhere else..

**Comment by Mothblocks** [comment]
no not just because its a goon thing

**Comment by necromanceranne** [comment]
would you prefer stamina obfuscated in some way? It's probably one of the more difficult damage types to get a sense of how effective it is. (not helped that there exist no external visual elements to tell you how impactful your damage is actually being)

**Comment by DrDiasyl** [comment]
>  would you prefer stamina obfuscated in some way? It's probably one of the more difficult damage types to get a sense of how effective it is. (not helped that there exist no external visual elements to tell you how impactful your damage is actually being)
>  
>  (although this is entirely spitballing, but I also wonder if that contributes to player perception that stamina is a weak damage type despite the numbers indicating the opposite)

Exactly, this is why I want to make a PR so stamina damage will give external signs (Screaming, crying, moaning, shaking depending on severity)

**Comment by jacklecroy** [comment]
> Probably because it reminds you of something.. same.. I think I seen this lighting symbol somewhere else..

It's from hippie.

> would you prefer stamina obfuscated in some way? It's probably one of the more difficult damage types to get a sense of how effective it is. (not helped that there exist no external visual elements to tell you how impactful your damage is actually being)

I already say in the OP, you still can't get an exact value from this. Personally, I'm not a fan of having to use cues like that for something that has such an immediate threat on you. It makes more sense for how viruses are. Either way, until something like that is added, a meter like this will be very helpful--especially to new players.

**Comment by LemonInTheDark** [comment]
Could you link the pr the sprites came from?

**Comment by jacklecroy** [comment]
> Could you link the pr the sprites came from?

https://github.com/HippieStationArchive/HippieStation13-1/pull/2464

**Comment by LemonInTheDark** [comment]
Thank you, I appreciate it

**Comment by LemonInTheDark** [comment]
> i don't know why, but i feel off about this

Is it because stamina crit takes your current health into account when triggering?

File: code/modules/mob/living/carbon/carbon.dm [hunk 1]
```diff
@@ -768,6 +768,29 @@
 		else
 			hud_used.healths.icon_state = "health7"
 
+/mob/living/carbon/update_stamina_hud(shown_stamina_amount)
+	if(!client || !hud_used || !hud_used?.stamina)
```
Comment (ShizCalev) [inline code/modules/mob/living/carbon/carbon.dm:772]:
```suggestion
	if(!client || !hud_used?.stamina)
```

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Comment by Mothblocks** [comment]
We talked about this more in maintainerbus and whatever worries I have are sated

**Review by Mothblocks — APPROVED** [review]
Codes fine now, I'll let art-tainers act on the sprites.

**Pull request merged**